tests/setup_test.go

```diff
@@ -5653,3 +5655,195 @@ func TestRunEventsNotification(t *testing.T) {
+			expectedGiteaStatusState: gitea.StatusFailure,
+			expectedGiteaDescription: "The run failed",
+			expectedGiteaContext:     "agola/project01/run01",
+		},
+		{
+			name: "test run setup config error",
+			config: `
+				{
+				  runserror:
+				}
+				`,
+			expectedRunResult:        rstypes.RunResultUnknown,
+			expectedRunPhase:         rstypes.RunPhaseSetupError,
+			expectedGiteaStatusState: gitea.StatusError,
+			expectedGiteaDescription: "The run encountered an error",
+			expectedGiteaContext:     "agola/project01/Setup Error",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dir := t.TempDir()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			sc := setup(ctx, t, dir, withGitea(true))
+			defer sc.stop()
+
+			giteaToken, tokenUser01 := createLinkedAccount(ctx, t, sc.gitea, sc.config)
+
+			giteaAPIURL := fmt.Sprintf("http://%s:%s", sc.gitea.HTTPListenAddress, sc.gitea.HTTPPort)
+
+			giteaClient, err := gitea.NewClient(giteaAPIURL, gitea.SetToken(giteaToken))
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+
+			gwClient := gwclient.NewClient(sc.config.Gateway.APIExposedURL, tokenUser01)
+
+			giteaRepo, project := createProject(ctx, t, giteaClient, gwClient)
+
+			push(t, tt.config, giteaRepo.CloneURL, giteaToken, "commit", false)
+
+			_ = testutil.Wait(30*time.Second, func() (bool, error) {
+				runs, _, err := gwClient.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+				if err != nil {
+					return false, nil
+				}
+
+				if len(runs) == 0 {
+					return false, nil
+				}
+				if runs[0].Phase != tt.expectedRunPhase {
+					return false, nil
+				}
+
+				return true, nil
+			})
+
+			runs, _, err := gwClient.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+			if len(runs) != 1 {
+				t.Fatalf("expected %d run got: %d", 1, len(runs))
+			}
+
+			if runs[0].Phase != tt.expectedRunPhase {
+				t.Fatalf("expected run phase %q, got %q", tt.expectedRunPhase, runs[0].Phase)
+			}
+			if runs[0].Result != tt.expectedRunResult {
+				t.Fatalf("expected run result %q, got %q", tt.expectedRunResult, runs[0].Result)
+			}
+
+			_ = testutil.Wait(30*time.Second, func() (bool, error) {
+				combinedStatus, _, err := giteaClient.GetCombinedStatus(agolaUser01, giteaRepo.Name, "master")
+				if err != nil {
+					return false, nil
+				}
+
+				if combinedStatus.State != tt.expectedGiteaStatusState {
+					return false, nil
+				}
+
+				return true, nil
+			})
+
+			// it has been copied from the notification service
+			u, err := url.Parse(sc.config.Notification.WebExposedURL + "/run")
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+			q := url.Values{}
+			q.Set("projectref", project.ID)
+			q.Set("runnumber", strconv.FormatUint(runs[0].Number, 10))
+			u.RawQuery = q.Encode()
+			targetURL := u.String()
```

Comment:
outside the tests range but inside the TestFunction define an inline function and use it here.